core/org.wso2.carbon.utils/src/main/java/org/wso2/carbon/utils/security/KeystoreUtils.java

```diff
@@ -0,0 +1,94 @@
+package org.wso2.carbon.utils.security;
+
+import org.wso2.carbon.utils.CarbonUtils;
+
+/**
+ * A collection of key store and trust store related utility methods.
+ */
+public class KeystoreUtils {
+
+    /**
+     * A collection of file type extensions against the store file type.
+     */
+    public enum StoreFileType {
+        JKS(".jks"),
+        PKCS12(".p12");
+
+        private final String extension;
+
+        StoreFileType(String extension) {
+
+            this.extension = extension;
+        }
+
+        /**
+         * Get the file extension for give store file type (ex: .jks, .p12).
+         *
+         * @return File extension.
+         */
+        public static String getFileTypeByExtension(String fileType) {
+
+            return StoreFileType.valueOf(fileType).extension;
+        }
+    }
+
+    /**
+     * Retrieve keystore file location.
+     *
+     * @return File location.
+     */
+    public static String getKeyStoreFileLocation() {
+
+        return CarbonUtils.getServerConfiguration().getFirstProperty("Security.KeyStore.Location");
+    }
+
+    /**
+     * Retrieve keystore file type (ex: JKS, PKCS12).
+     *
+     * @return File type.
+     */
+    public static String getKeyStoreFileType() {
+
+        return CarbonUtils.getServerConfiguration().getFirstProperty("Security.KeyStore.Type");
```

Comment:
I think we should default to PKCS in case `CarbonUtils.getServerConfiguration().getFirstProperty("Security.KeyStore.Type")` returns a null.

Same with the trustsore method

Reply:
It will not return a null as we set an default value from default.json. However, add a validation file type validation logic with 84a06b4ccf37ef09e4e00fb05cccf895343fe69a